samplers/jaegerremote/sampler_remote.go

```diff
@@ -273,15 +273,20 @@ type httpSamplingStrategyFetcher struct {
 }
 
 func newHTTPSamplingStrategyFetcher(serverURL string) *httpSamplingStrategyFetcher {
-	customTransport := http.DefaultTransport.(*http.Transport).Clone()
-	customTransport.ResponseHeaderTimeout = defaultRemoteSamplingTimeout
-
-	return &httpSamplingStrategyFetcher{
+	fetcher := &httpSamplingStrategyFetcher{
 		serverURL: serverURL,
-		httpClient: http.Client{
-			Transport: customTransport,
-		},
 	}
+
+	if customTransport, ok := http.DefaultTransport.(*http.Transport); ok {
```

Comment:
If I understand correctly. If `http.DefaultTransport` is NOT `*http.Transport` then the request send by the fetcher would NOT have any timeout. If I am correct, then I do not think it is acceptable.

Maybe we should always set `http.Client.Timeout` instead of setting `customTransport.ResponseHeaderTimeout`? If my thinking is correct then would we even need a custom transport?

CC @yurishkuro

Reply:
> Maybe we should always set http.Client.Timeout instead of setting customTransport.ResponseHeaderTimeout?

I like this idea, updated.

Reply:
@yurishkuro please double-check my idea. I do not trust myself 😉